deepchem/feat/huggingface_featurizer.py

```diff
@@ -0,0 +1,40 @@
+from typing import TYPE_CHECKING
+from deepchem.feat.base_classes import Featurizer
+
+if TYPE_CHECKING:
+    import transformers
+
+
+class HuggingFaceFeaturizer(Featurizer):
+    """Wrapper class that wraps HuggingFace tokenizers as DeepChem featurizers
+
+    The `HuggingFaceFeaturizer` wrapper provides a wrapper
+    around Hugging Face tokenizers allowing them to be used as DeepChem
+    featurizers. This might be useful in scenarios where user needs to use
+    a hugging face tokenizer when loading a dataset.
+
+    Example
+    -------
+    >>> from deepchem.feat import HuggingFaceFeaturizer
+    >>> from transformers import RobertaTokenizerFast
+    >>> hf_tokenizer = RobertaTokenizerFast.from_pretrained("seyonec/PubChem10M_SMILES_BPE_60k")
+    >>> featurizer = HuggingFaceFeaturizer(tokenizer=hf_tokenizer)
+    >>> result = featurizer.featurize(['CC(=O)C'])
+    """
+
+    def __init__(
+        self,
+        tokenizer: 'transformers.tokenization_utils_fast.PreTrainedTokenizerFast'
```

Comment:
I am not familiar with this notation. The type of the tokenizer is set to a string? Is this something using in type checking to avoid a circular import?

Reply:
`transformers` is an expensive module for import (takes about 3-4 seconds in my machine). 

In the above module, we need to import the transformers module only for type checking purpose. Hence, I have enclosed it in strings so that it is hided from the interpreter runtime, thereby reducing import time of the module. During type check, it gets imported because the variable `TYPE_CHECKING` (used [here](https://github.com/deepchem/deepchem/blob/607baf59c6b256c488ba8ed2d4088b46a1456205/deepchem/feat/huggingface_featurizer.py#L27)) will be True.

I found this usage from python docs ([ref](https://docs.python.org/3/library/typing.html#typing.TYPE_CHECKING)) and I have also seen a similar usage in other projects.